src/execution.rs

```diff
@@ -50,6 +51,23 @@ pub struct Execution {
     pub use_hooks: bool,
```

Comment:
If you extract the program, then the `ExecutionSnapshot` becomes just a clone of this. Maybe we could move the program out and turn `ExecutionSnapshot` in a simple wrapper?